Documentation/DebuggingTechniques.md

```diff
@@ -0,0 +1,39 @@
+# Debugging Techniques
+
+This document contains list of debugging techniques and infrastructure that helpful for debugging Reactive Cocoa applications.
+
+#### Unscrambling Swift compiler errors
+
+Type inferrence could be a source of hard-to-debug compiler errors. There are two potential places to be wrong when type inferrence used:
+
+1. Definition of type inferred variable
+2. Consumption of type inferred variable
+
+In both cases errors are related to incorrect assumptions about type. Such issues are common for a ReactiveCocoa applications as it is all about operations over data and related types. With a current state of a Swift compiler type related errors could be really misleading, especially when error happens in the middle of a signal chain. 
```

Comment:
~~With a current state of a Swift compiler type related errors could be really misleading,~~
The current state of the Swift compiler can cause misleading type errors,